### ISSUE TITLE
[BUG] dont rewrite user headers

### ISSUE THREAD

**Issue by lolyinseo** [lexiforest/curl_cffi#408]
```        # Add content-type if missing
        if json is not None:
            _update_header_line(header_lines, "Content-Type", "application/json")
        if isinstance(data, dict) and method != "POST":
            _update_header_line(header_lines, "Content-Type", "application/x-www-form-urlencoded")
        if isinstance(data, (str, bytes)):
            _update_header_line(header_lines, "Content-Type", "application/octet-stream")
```

You dont "Add content-type if missing" - you just rewrite it. Of course it breaks everything

**Comment by lexiforest** [comment]
Duplicate of #395, should be fixed in v0.7.3.

**Comment by lolyinseo** [comment]
> Duplicate of #395, should be fixed in v0.7.3.

Nope. In this code

```
        # Add content-type if missing
        if json is not None:
            _update_header_line(header_lines, "Content-Type", "application/json", replace=True)
        if isinstance(data, dict) and method != "POST":
            _update_header_line(header_lines, "Content-Type", "application/x-www-form-urlencoded")
        if isinstance(data, (str, bytes)):
            _update_header_line(header_lines, "Content-Type", "application/octet-stream")
```

you are REPLACE Content-Type

correct code 

```
        if h.get("Content-Type") is None:

            # Add content-type if missing
            if json is not None:
                _update_header_line(header_lines, "Content-Type", "application/json")
            if isinstance(data, dict) and method != "POST":
                _update_header_line(header_lines, "Content-Type", "application/x-www-form-urlencoded")
            if isinstance(data, (str, bytes)):
                _update_header_line(header_lines, "Content-Type", "application/octet-stream")
```

**Comment by lexiforest** [comment]
Note the `replace` parameter:

https://github.com/lexiforest/curl_cffi/blob/06a161457fd4d4fb0c1d1770590ca701c2766482/curl_cffi/requests/session.py#L484-L489

https://github.com/lexiforest/curl_cffi/blob/06a161457fd4d4fb0c1d1770590ca701c2766482/curl_cffi/requests/session.py#L186-L187
